src/main/java/com/pokegoapi/auth/GoogleAutoCredentialProvider.java

```diff
@@ -21,63 +21,79 @@
 	private static String GOOGLE_LOGIN_APP = "com.nianticlabs.pokemongo";
 	private static String GOOGLE_LOGIN_CLIENT_SIG = "321187995bc7cdc2b5fc91b11a96e2baa8602c62";
 
+	private final Gpsoauth gpsoauth;
 	private final String username;
-	private final String password;
+	private TokenInfo tokenInfo;
 
-	private AuthToken authToken;
-	private AuthInfo authInfo;
 
 	/**
 	 * Constructs credential provider using username and password
 	 *
 	 * @param username - google username
 	 * @param password - google password
-	 * @throws LoginFailedException - login failed possibly due to invalid credentials
+	 * @throws LoginFailedException  - login failed possibly due to invalid credentials
 	 * @throws RemoteServerException - some server/network failure
 	 */
-	public GoogleAutoCredentialProvider(String username, String password)
+	public GoogleAutoCredentialProvider(OkHttpClient httpClient, String username, String password)
 			throws LoginFailedException, RemoteServerException {
+		this.gpsoauth = new Gpsoauth(httpClient);
 		this.username = username;
-		this.password = password;
-		login(username, password);
+		this.tokenInfo = login(username, password);
 	}
 
-	private void login(String username, String password) throws RemoteServerException, LoginFailedException {
+	private TokenInfo login(String username, String password) throws RemoteServerException, LoginFailedException {
 		try {
-			OkHttpClient httpClient = new OkHttpClient();
-			Gpsoauth gpsoauth = new Gpsoauth(httpClient);
-			this.authToken = gpsoauth.login(
-					username,
-					password,
-					GOOGLE_LOGIN_ANDROID_ID,
-					GOOGLE_LOGIN_SERVICE,
-					GOOGLE_LOGIN_APP,
-					GOOGLE_LOGIN_CLIENT_SIG
-			);
+			String masterToken = gpsoauth.performMasterLoginForToken(username, password, GOOGLE_LOGIN_ANDROID_ID);
+			AuthToken authToken = gpsoauth.performOAuthForToken(username, masterToken, GOOGLE_LOGIN_ANDROID_ID,
+					GOOGLE_LOGIN_SERVICE, GOOGLE_LOGIN_APP, GOOGLE_LOGIN_CLIENT_SIG);
+			return new TokenInfo(authToken, masterToken);
 		} catch (IOException e) {
 			throw new RemoteServerException(e);
 		} catch (Gpsoauth.TokenRequestFailed e) {
 			throw new LoginFailedException(e);
 		}
 	}
 
+	private TokenInfo refreshToken(String username, String refreshToken) throws RemoteServerException, LoginFailedException {
```

Comment:
Line is longer than 120 characters (found 123). [LineLength]

This causes a build failure, but is easily resolved by renaming to `refToken`